src/createAnimatedComponent/createAnimatedComponent.tsx

```diff
@@ -254,61 +255,63 @@ export function createAnimatedComponent(
           has('workletEventHandler', prop) &&
           prop.workletEventHandler instanceof WorkletEventHandler
         ) {
-          prop.workletEventHandler.unregisterFromEvents();
-        }
-      }
-    }
-
-    _detachStyles() {
-      if (IS_WEB && this._styles !== null) {
-        for (const style of this._styles) {
-          style.viewsRef.remove(this);
-        }
-      } else if (this._viewTag !== -1 && this._styles !== null) {
-        for (const style of this._styles) {
-          style.viewDescriptors.remove(this._viewTag);
-        }
-        if (this.props.animatedProps?.viewDescriptors) {
-          this.props.animatedProps.viewDescriptors.remove(this._viewTag);
-        }
-        if (isFabric()) {
-          removeFromPropsRegistry(this._viewTag);
+          prop.workletEventHandler.unregisterFromEvents(
+            this._getViewInfo().viewTag as number
+          );
         }
       }
     }
 
-    _reattachNativeEvents(
+    _updateNativeEvents(
       prevProps: AnimatedComponentProps<InitialComponentProps>
     ) {
+      let previousHandler: IWorkletEventHandler<Event> | undefined;
+      let newHandler: IWorkletEventHandler<Event> | undefined;
+
+      // Get previous handler
```

Comment:
Here you assume that there can be only one handler in a component. What if we had two separate handlers for `onScroll` and `onMomentumBegin`? This function needs a lot of improvement 😈

Reply:
@tjzel  This PR aims to allow multiple components for one handler, not the other way (which also wasn't there before). However, I might try adding it if You feel like it is important.

Reply:
I'm pretty sure you could pass two handlers to a single component before.